### PR TITLE
fix: yat grapheme validation

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fixes y.at validation, currently failing for some emojis with the current logic.
The tl;dr on that is that `/^\p{Extended_Pictographic}{1,5}$/u` doesn't test/match what we would naively think it does.

- `'🦊❤️😎❤️'.match(/\p{Extended_Pictographic}/gu)` // `["🦊","❤","😎","❤"]`  (notice the removed caret and dollarsign as well as the 1,5 multiplicator)
- `'🦊❤️😎❤️'.match(/\p{Extended_Pictographic}+/gu)` // `[ "🦊❤", "😎❤" ]` (notice the + multiplicator, and how this now creates two groups of two "emojis")
- `'🦊❤️😎❤️'.match(/^\p{Extended_Pictographic}{1,5}^/gu)` // `null`

Fixed by using the `grapheme-splitter` as recommended by Gabriel from y.at, which ensures we split characters by graphemes e.g "user-visible characters", as opposed to actual characters, which emojis might be made of many.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None, worst case scenario this is still broken

## Testing

- Ensure valid y.ats now pass validation
- Invalid y.ats (not registered, or with some non-emojis characters somewhere in the string) are still failing validation

### Engineering

☝🏽 

### Operations

☝🏽 

## Screenshots (if applicable)

image
image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3909** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3909" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3909?utm_source=stack-comment).

**Review by 0xdef1cafe — APPROVED** [review]
Very naice
